core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -2537,6 +2569,38 @@ oppia.factory('autosaveInfoModalsService', [
         });
 
         _isModalOpen = true;
+      },
+      showLostChangesModal: function(lostChanges, explorationId) {
+        $modal.open({
+          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+            '/pages/exploration_editor/' +
+            'lost_changes_modal.html'),
+          // Prevent modal from closing when the user clicks outside it.
+          backdrop: 'static',
+          controller: ['$scope', '$modalInstance', function(
+            $scope, $modalInstance) {
+            // When the user clicks on discard changes button, signal backend
+            // to discard the draft and reload the page thereafter.
+            $scope.close = function() {
+              LocalStorageService.removeExplorationDraft(explorationId);
+              $modalInstance.dismiss('cancel');
+            };
+
+            $scope.hasLostChanges = (lostChanges && lostChanges.length > 0);
```

Comment:
Not sure I understand ... why would we even show this modal if there are no lost changes?